data/generateReviews.js

```diff
@@ -69,10 +77,14 @@ function generateReviews(db, numReviews, numUsers, numNeighborhoods, writer, enc
 }
 
 // generate 100M reviews with 10M users and 10M neighborhoods
-generateReviews('sql', 100000000, 10000000, 10000000, writeReviewsSQL, 'utf-8', () => {
-  writeReviewsSQL.end();
-});
+async function generateCSV() {
+  await generateReviews('sql', 100000000, 10000000, 10000000, writeReviewsSQL, 'utf-8', () => {
```

Comment:
To my knowledge, the fs.write, fs.writeFile, etc. functions in node.js are asynchronous as node.js itself is non-blocking/asynchronous across the board unless specified to be synchronous seen with [fs.writeFileSync](https://nodejs.org/api/fs.html#fs_fs_writefilesync_file_data_options) and [fs.writeSync](https://nodejs.org/api/fs.html#fs_fs_writesync_fd_buffer_offset_length_position). Because of this, I think the async/await specification for this function isn't needed as its already asynchronous.

Reply:
yeah fs.createWriteStream is asynchronous but I wanted to generate each csv synchronously. So the next function will only execute once the promise is resolved.
Not sure if this is bad practice to have it become synchronous but I wanted to dedicate all my CPU usage for generating one file at a time. Because my computer is OLD